crates/core/src/factories/core_fetch/test/request_queue.rs

```diff
@@ -32,10 +35,20 @@ fn make_mock_transport(
             assert_eq!(space, SPACE_ID);
             assert_eq!(module, crate::factories::core_fetch::MOD_NAME);
             Box::pin({
-                let value = requests_sent.clone();
+                let requests_sent = requests_sent.clone();
```

Comment:
This is where I'm wondering if we've not got enough in core to be able to test with the in-memory transport and just observe the results, rather than the inner messages?

Reply:
Yes, likely so. An in-memory transport could just keep a hash map of registered handlers and do the forwarding. What do you mean by observe the results? How should the transport process inner messages to achieve the results?

Reply:
I thought at some point there were tests here checking that the right outgoing messages were sent in response but maybe that never worked through the transport :) That's not a problem if the tests aren't depending on getting hold of the messages and can just stub the receiver of the message to check that it arrived

There already is an in-memory transport, the coretransport type does that

Reply:
The following file checks outgoing responses. Is the problem there?

Reply:
Probably not, I was struggling to find what I was looking for because the diff is quite noisy. It was more of a general question. This looks fine as is :)